### PR TITLE
Clean up memory allocated when reading messages in H5Dlayout on error (#2602)

### PR DESCRIPTION
**Pull request by glennsong09**

### CODE REVIEW

File: src/H5Dlayout.c [hunk 1]
```diff
@@ -588,9 +588,11 @@ H5D__layout_oh_create(H5F_t *file, H5O_t *oh, H5D_t *dset, hid_t dapl_id)
 herr_t
 H5D__layout_oh_read(H5D_t *dataset, hid_t dapl_id, H5P_genplist_t *plist)
 {
-    htri_t  msg_exists;              /* Whether a particular type of message exists */
-    hbool_t layout_copied = FALSE;   /* Flag to indicate that layout message was copied */
-    herr_t  ret_value     = SUCCEED; /* Return value */
+    htri_t  msg_exists;                      /* Whether a particular type of message exists */
+    hbool_t dcpl_cache_pline_copied = FALSE; /* Flag to indicate that dcpl_cache.pline's message was copied */
```
Comment (derobins) [inline src/H5Dlayout.c]: It's a small thing, but I would make the naming scheme consistent, probably by dropping the dcpl_cache from the name of this identifier.

**Review by derobins — APPROVED** [review]
I would consider making the small change I requested, but otherwise this looks good.

**Comment by glennsong09** [comment]
I will make the change.

**Comment by jhendersonHDF** [comment]
Also, please add an entry in release_docs/RELEASE.txt with a short description of the issue and fix. Otherwise this looks good to me.

**Comment by derobins** [comment]
> Also, please add an entry in release_docs/RELEASE.txt with a short description of the issue and fix. Otherwise this looks good to me.

Oh yeah, forgot to emphasize that. Should definitely do that.

File: release_docs/RELEASE.txt [hunk 2]
```diff
@@ -151,6 +151,12 @@ Bug Fixes since HDF5-1.13.3 release
 ===================================
     Library
     -------
+    - Fixed memory leaks that could occur when reading a dataset from a malformed file
+
+      When attempting to read layout, pline, and efl information for a dataset, memory leaks could occur if attempting to read pline/efl information threw an error, which is due to the memory that was allocated for pline and efl not being properly cleaned up on error.
```
Comment (derobins) [inline release_docs/RELEASE.txt]: This should wrap on 75-ish characters

**Review by lrknox — APPROVED** [review]

**Pull request merged**